src/mlagility/analysis/status.py

```diff
@@ -16,96 +16,147 @@ def update(models_found: Dict[str, ModelInfo]) -> None:
+
+                print_workload(
+                    model_info,
+                    workload_hash,
+                    print_file_name,
+                    workload_idx=workload_idx,
+                    model_visited=model_visited,
+                )
+                model_visited = True
+                workload_idx += 1
+
+                if print_file_name:
+                    script_names_visited.append(model_info.script_name)
+
+                recursive_print(
+                    models_found,
+                    parent_model_hash=model_hash,
+                    parent_workload_hash=workload_hash,
+                    script_names_visited=script_names_visited,
+                )
+
+
+def print_workload(
+    model_info: ModelInfo,
+    workload_hash: Union[str, None],
+    print_file_name: bool = False,
+    workload_idx: int = 0,
+    model_visited: bool = False,
 ) -> None:
     """
     Print information about a given model or submodel
     """
+    workload = model_info.workloads[workload_hash]
     ident = "\t" * (2 * model_info.depth + 1)
     if print_file_name:
         print(f"{model_info.script_name}.py:")
-    printing.log(f"{ident}{model_info.name} ")
 
-    # Show the number of times the model has been executed
-    # Only show the execution time if we are not running benchit() as this
-    # impacts time measurement.
-    if model_info.exec_time == 0 or model_info.build_model:
+    if workload.exec_time == 0 or model_info.build_model:
         exec_time = ""
     else:
-        exec_time = f" - {model_info.exec_time:.2f}s"
-    printing.logn(
-        f"(executed {model_info.executed}x{exec_time})",
-        c=printing.Colors.OKGREEN,
-    )
+        exec_time = f" - {workload.exec_time:.2f}s"
+
+    if model_info.depth == 0 and len(model_info.workloads) > 1:
+        if not model_visited:
+            printing.logn(f"{ident}{model_info.name}")
+    else:
+        printing.log(f"{ident}{model_info.name}")
+        printing.logn(
+            f" (executed {workload.executed}x{exec_time})",
+            c=printing.Colors.OKGREEN,
+        )
+
+    if (model_info.depth == 0 and not model_visited) or (model_info.depth != 0):
+        if model_info.depth == 0:
+            if model_info.model_type == build.ModelType.PYTORCH:
+                print(f"{ident}\tModel Type:\tPytorch (torch.nn.Module)")
+            elif model_info.model_type == build.ModelType.KERAS:
+                print(f"{ident}\tModel Type:\tKeras (tf.keras.Model)")
+
+        # Display class of found model and the file where it was found
+        model_class = type(model_info.model)
+        print(f"{ident}\tClass:\t\t{model_class.__name__} ({model_class})")
+        if model_info.depth == 0:
+            print(f"{ident}\tLocation:\t{model_info.file}, line {model_info.line}")
+
+        # Converting number of parameters to MB assuming 2 bytes per parameter
+        model_size = model_info.params * 2 / (1024 * 1024)
```

Comment:
nit: I never noticed before that the `2` was hardcoded here. This is a pretty spicy assumption considering that if we run `--analyze-only` on a model, odds are that the model is going to be fp32. 

Might be safer to just remove the `({model_size} MB)` part from the printout at this juncture. We could populate that info once we have an ONNX file and the dtype of the weights are locked.

Reply:
I agree that this could be confusing and should be addressed.
Just created an issue for that to avoid scope creep here.
https://github.com/groq/mlagility/issues/330